src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPOpenIdAuthenticator.java

```diff
@@ -0,0 +1,328 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.config.RequestConfig;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManagerBuilder;
+import org.apache.hc.client5.http.io.HttpClientConnectionManager;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.auth.HTTPAuthenticator;
+import org.opensearch.security.filter.SecurityRequest;
+import org.opensearch.security.filter.SecurityResponse;
+import org.opensearch.security.user.AuthCredentials;
+
+import com.amazon.dlic.auth.http.jwt.AbstractHTTPJwtAuthenticator;
+import com.amazon.dlic.util.SettingsBasedSSLConfigurator;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.APPLICATION_JSON;
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.APPLICATION_JWT;
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.AUTHORIZATION_HEADER;
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.CLIENT_ID;
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.ISSUER_ID_URL;
+import static com.amazon.dlic.auth.http.jwt.keybyoidc.OpenIdConstants.SUB_CLAIM;
+
+public class HTTPOpenIdAuthenticator implements HTTPAuthenticator {
+
+    private final static Logger log = LogManager.getLogger(HTTPOpenIdAuthenticator.class);
+    private final Settings settings;
+    private final Path configPath;
+    private final int requestTimeoutMs = 10000;
+    private final SettingsBasedSSLConfigurator.SSLConfig sslConfig;
+    private final String userinfo_endpoint;
+    private volatile HTTPJwtKeyByOpenIdConnectAuthenticator openIdJwtAuthenticator;
+
+    public HTTPOpenIdAuthenticator(Settings settings, Path configPath) throws Exception {
+        this.settings = settings;
+        this.configPath = configPath;
+        this.sslConfig = getSSLConfig(settings, configPath);
+        userinfo_endpoint = settings.get("userinfo_endpoint");
+        openIdJwtAuthenticator = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, configPath);
+    }
+
+    public HTTPJwtKeyByOpenIdConnectAuthenticator getOpenIdJwtAuthenticator() {
+        if (openIdJwtAuthenticator == null) {
```

Comment:
I made this for the case where we want to access the wrapped jwtAuthenticator. If the overall Authenticator is not initialized, we would want to be able to access the internal authenticator for testing I believe. The synchronized is meant to prevent the creation of multiple authenticators when `getOpenIdJwtAuthenticator` is called multiple times. We don't want to create several instances of the `jwtOpenIdAuthenticator` so this makes sure we only ever make one. 

If you think there is a better way of doing this let me know.

Reply:
I am going to remove this since apparently this pattern is not recommended in modern coding practices ...